examples/vl3_basic/vl3-nse/cmd/service_registry.go

```diff
@@ -0,0 +1,154 @@
+	connection *grpc.ClientConn
+}
+
+func (s *ServiceRegistryImpl) RegisterWorkload(clusterName, podName, name, seviceName, connDom string, ipAddr []string, ports []int32) error {
+	workloadIdentifier := &serviceregistry.WorkloadIdentifier{
+		Cluster:             clusterName,
+		PodName:             podName,
+		Name:                name,
+	}
+
+	workload := &serviceregistry.Workload{
+		Identifier:          workloadIdentifier,
+		IPAddress:           ipAddr,
+	}
+
+	workloads := []*serviceregistry.Workload{workload}
+	serviceWorkload := &serviceregistry.ServiceWorkload{
+		ServiceName:         seviceName,
+		ConnectivityDomain:  connDom,
+		Workloads:           workloads,
+		Ports:               ports,
+	}
+
+	logrus.Infof("Sending workload register request: %v", serviceWorkload)
+	_, err := s.registryClient.RegisterWorkload(context.Background(), serviceWorkload)
+	if err != nil {
+		logrus.Errorf("service registration not successful: %v", err)
+	}
+
+	return err
+}
+
+func (s *ServiceRegistryImpl) RemoveWorkload(clusterName, podName, name, seviceName, connDom string, ipAddr []string, ports []int32) error {
+	workloadIdentifier := &serviceregistry.WorkloadIdentifier{
+		Cluster:             clusterName,
+		PodName:             podName,
+		Name:                name,
+	}
+
+	workload := &serviceregistry.Workload{
+		Identifier:          workloadIdentifier,
+		IPAddress:           ipAddr,
+	}
+
+	workloads := []*serviceregistry.Workload{workload}
+	serviceWorkload := &serviceregistry.ServiceWorkload{
+		ServiceName:         seviceName,
+		ConnectivityDomain:  connDom,
+		Workloads:           workloads,
+		Ports:               ports,
+	}
+
+	logrus.Infof("Sending workload remove request: %v", serviceWorkload)
+	_, err := s.registryClient.RegisterWorkload(context.Background(), serviceWorkload)
+	if err != nil {
+		logrus.Errorf("service removal not successful: %v", err)
+	}
+
+	return err
+}
+
+func (s *ServiceRegistryImpl) Stop() {
+	s.connection.Close()
+}
+
+func NewServiceRegistry(addr string) (ServiceRegistry, ServiceRegistryClient, error) {
+	conn, err := grpc.Dial(addr, grpc.WithInsecure())
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to connect to ipam server: %v", err)
+	}
+
+	registryClient := serviceregistry.NewRegistryClient(conn)
+	serviceRegistry := ServiceRegistryImpl{registryClient: registryClient, connection: conn}
+
+	return &serviceRegistry, &serviceRegistry, nil
+}
+
+func processPortsFromLabel(portLabel, separator string) ([]int32, error) {
+
+	ports := strings.Split(portLabel, separator)
+	servicePorts := []int32{}
+	for _, port := range ports {
+		portToInt, err := strconv.ParseInt(port, 10, 32)
+		if err != nil {
+			return nil, err
+		}
+		servicePorts = append(servicePorts, int32(portToInt))
+	}
+
+	return servicePorts, nil
+}
+
+func processWorkloadIps(workloadIps, separator string) []string {
+
+	ips := strings.Split(workloadIps, separator)
+	serviceIps := []string{}
+	for _, ip := range ips {
```

Comment:
vl3-nse/cmd/service_registry.go:130:2: S1011: should replace loop with `serviceIps = append(serviceIps, ips...)` (gosimple)
        for _, ip := range ips {
        ^


Run a linter with the same command we use in cnns-nsr:
user@ubuntu:~/work/vladmoisescu/examples/examples/vl3_basic$ go fmt ./...&& CGO_ENABLED=0 goimports -w -d . `find . -type f -name '*.go' -not -name '*.pb.go'` && GO111MODULE=on CGO_ENABLED=0 LOG_LEVEL=error /home/user/go/bin/golangci-lint run  ./... && CGO_ENABLED=0 go test -cover -tags unit ./...

Reply:
Done